src/main/java/org/opensrp/web/acl/EventPermissionEvaluator.java

```diff
@@ -0,0 +1,73 @@
+package org.opensrp.web.acl;
+
+import org.opensrp.domain.Event;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Component;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+@Component
+public class EventPermissionEvaluator extends BasePermissionEvaluator<Event> {
+
+	public boolean hasPermission(Authentication authentication, Event targetDomainObject) {
+		return hasPermissionOnEvent(authentication, targetDomainObject.getLocationId());
+	}
+
+	private boolean hasPermissionOnEvent(Authentication authentication, String identifier) {
+		/* @formatter:off */
+		return getAssignedLocations(authentication.getName())
+				.stream()
+				.anyMatch(a -> a.getPlanId().equals(identifier));
+		/* @formatter:on */
+	}
```

Comment:
Checking permission using event Id will not be possible, pass the actual event object 

Use this logic 
https://github.com/OpenSRP/opensrp-server-web/pull/405/files#r442872793